### PR TITLE
Volume, CeilingHeight, FloorArea for Space object, try 2

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

Follow same behavior as ThermalZone: check field for hard set double value, and FT only if populated.

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by shorowit** [comment]
I haven't tested all possible situations, but can confirm that this fixes the regression I saw for one situation (1:1 space:zone, with zone volume hardsized). OS 3.5 now gives identical results to OS 3.4.

File: src/energyplus/ForwardTranslator/ForwardTranslateSpace.cpp [hunk 1]
```diff
@@ -86,14 +87,22 @@ namespace energyplus {
...
-    idfObject.setDouble(SpaceFields::CeilingHeight, modelObject.ceilingHeight());
+    if (modelObject.getDouble(OS_SpaceFields::CeilingHeight)) {
+      idfObject.setDouble(SpaceFields::CeilingHeight, modelObject.getDouble(SpaceFields::CeilingHeight).get());
+    }
 
     // Volume
-    idfObject.setDouble(SpaceFields::Volume, modelObject.volume());
+    if (modelObject.getDouble(OS_SpaceFields::Volume)) {
+      idfObject.setDouble(SpaceFields::Volume, modelObject.getDouble(OS_SpaceFields::Volume).get());
+    }
```
Comment (jmarrec) [inline src/energyplus/ForwardTranslator/ForwardTranslateSpace.cpp:100]:
```suggestion
    if (!modelObject.isVolumeDefaulted()) {
      idfObject.setDouble(SpaceFields::Volume, modelObject.volume());
    }
```

File: src/energyplus/ForwardTranslator/ForwardTranslateSpace.cpp [hunk 2]
```diff
@@ -86,14 +87,22 @@ namespace energyplus {
       idfObject.setString(SpaceFields::ZoneName, thermalZone->name().get());
     }
 
+    // for CeilingHeight, Volume, FloorArea: only FT if hard set (same logic as for ThermalZone)
+
     // CeilingHeight
-    idfObject.setDouble(SpaceFields::CeilingHeight, modelObject.ceilingHeight());
+    if (modelObject.getDouble(OS_SpaceFields::CeilingHeight)) {
+      idfObject.setDouble(SpaceFields::CeilingHeight, modelObject.getDouble(SpaceFields::CeilingHeight).get());
+    }
```
Comment (jmarrec) [inline src/energyplus/ForwardTranslator/ForwardTranslateSpace.cpp:95]:
```suggestion
    if (!modelObject.isCeilingHeightDefaulted()) {
      idfObject.setDouble(SpaceFields::CeilingHeight, modelObject.ceilingHeight());
    }
```

File: src/energyplus/ForwardTranslator/ForwardTranslateSpace.cpp [hunk 3]
```diff
@@ -86,14 +87,22 @@ namespace energyplus {
...
-    idfObject.setDouble(SpaceFields::Volume, modelObject.volume());
+    if (modelObject.getDouble(OS_SpaceFields::Volume)) {
+      idfObject.setDouble(SpaceFields::Volume, modelObject.getDouble(OS_SpaceFields::Volume).get());
+    }
 
     // FloorArea
-    idfObject.setDouble(SpaceFields::FloorArea, modelObject.floorArea());
+    if (modelObject.getDouble(OS_SpaceFields::FloorArea)) {
+      idfObject.setDouble(SpaceFields::FloorArea, modelObject.getDouble(OS_SpaceFields::FloorArea).get());
+    }
```
Comment (jmarrec) [inline src/energyplus/ForwardTranslator/ForwardTranslateSpace.cpp:105]:
```suggestion
    if (!modelObject.isFloorAreaDefaulted()) {
      idfObject.setDouble(SpaceFields::FloorArea, modelObject.floorArea());
    }
```

**Review by jmarrec — CHANGES_REQUESTED** [review]

**Pull request merged**